server/pkg/asset/preview_type_test.go

```diff
@@ -0,0 +1,166 @@
+package asset
+
+import (
+	"testing"
+
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPreviewType_PreviewTypeFrom(t *testing.T) {
+	tests := []struct {
+		Name     string
+		Expected struct {
+			TA   PreviewType
+			Bool bool
+		}
+	}{
+		{
+			Name: "IMAGE",
+			Expected: struct {
+				TA   PreviewType
+				Bool bool
+			}{
+				TA:   PreviewTypeIMAGE,
+				Bool: true,
+			},
+		},
+		{
+			Name: "GEO",
+			Expected: struct {
+				TA   PreviewType
+				Bool bool
+			}{
+				TA:   PreviewTypeGEO,
+				Bool: true,
+			},
+		},
+		{
+			Name: "GEO3D",
+			Expected: struct {
+				TA   PreviewType
+				Bool bool
+			}{
+				TA:   PreviewTypeGEO3D,
+				Bool: true,
+			},
+		},
+		{
+			Name: "MODEL3D",
+			Expected: struct {
+				TA   PreviewType
+				Bool bool
+			}{
+				TA:   PreviewTypeMODEL3D,
+				Bool: true,
+			},
+		},
+		{
+			Name: "undefined",
+			Expected: struct {
+				TA   PreviewType
+				Bool bool
+			}{
+				TA:   PreviewType(""),
+				Bool: false,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			res, ok := PreviewTypeFrom(tc.Name)
+			assert.Equal(t, tc.Expected.TA, res)
+			assert.Equal(t, tc.Expected.Bool, ok)
+		})
+	}
+}
+
+func TestPreviewType_PreviewTypeFromRef(t *testing.T) {
+	i := PreviewTypeIMAGE
+	g := PreviewTypeGEO
+	g3d := PreviewTypeGEO3D
+	m := PreviewTypeMODEL3D
+
+	tests := []struct {
+		Name     string
+		Input    *string
+		Expected *PreviewType
+	}{
+		{
+			Name:     "IMAGE",
+			Input:    getStrRef("IMAGE"),
```

Comment:
```suggestion
			Input:    lo.ToPtr("IMAGE"),
```

getStrRef is no longer needed. [lo](https://github.com/samber/lo) is a very helpful library.